linux/check_os_fullname/set_guestinfo_without_shortname.yml

```diff
@@ -64,6 +64,12 @@
     expected_guest_fullname: "SUSE openSUSE ({{ guest_os_bit }})"
   when: guest_os_ansible_distribution == "openSUSE Leap"
 
+- name: "Set expected guest id and full name for {{ vm_guest_os_distribution }}"
+  ansible.builtin.set_fact:
+    expected_guest_id: "asianux{{ guest_os_ansible_distribution_major_ver }}{{ expected_guest_id_suffix }}"
+    expected_guest_fullname: "MIRACLE LINUX {{ guest_os_ansible_distribution_major_ver }} ({{ guest_os_bit }})"
+  when: guest_os_ansible_distribution == "MIRACLE"
```

Comment:
You shall consider scenarios with different VMTools versions:
1. When vmtools_version is version('12.3.5', '<='), Miracle Linux 8.x will be recognized as other linux. So need to keep original logic for VMTools 12.3.5 or earlier.
2. When vmtools_version is version('12.3.5', '>'), you can set its guest id with asinux8_64bit and guest full name as MIRACLE LINUX 8 (64-bit)

Another scenario shall be noticed is when Miracle Linux 9.x is available, its expected guest id and guest full name is same as Miracle Linux 8.x on ESXi 8.x, as there is no Miracle Linux 9 (64-bit) selection in vSphere UI.